Library/Homebrew/download_strategy.rb

```diff
@@ -608,6 +610,60 @@ def fetch_release_metadata
   end
 end
 
+class ScpDownloadStrategy < AbstractFileDownloadStrategy
+  attr_reader :tarball_path, :temporary_path
+
+  def initialize(name, resource)
+    super
+    @tarball_path = HOMEBREW_CACHE/"#{name}-#{version}#{ext}"
+    @temporary_path = Pathname.new("#{cached_location}.incomplete")
+    parse_url_pattern
+  end
+
+  def parse_url_pattern
+    url_pattern = %r{scp://([^@]+@)?([^@:/]+)(:\d+)?/(\S+)}
```

Comment:
Can you supply an example test formula as a comment? I'll obviously have to edit it to match something I have `scp` access to but it would be helpful.

Reply:
Done!